WordPress/gutenberg
# Mobile - RichText - Update logic for the placeholder text color

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6182
- https://github.com/wordpress-mobile/WordPress-iOS/pull/21544

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6069

## What?
This PR updates the logic for the placeholder text colors for the RichText component and block-based themes.

## Why?
Previously, the same colors were displayed for both the placeholder text and the populated RichText component.

## How?
Opacity is added to the placeholder text color, utilizing the default text color derived from the theme settings.

The update also resolves a pre-existing issue where the empty placeholder displayed a smaller font size in instances where the current theme had a font size larger than the mobile editor's default. The font size now aligns with the Paragraph block's default size.

## Testing Instructions
**Precondition:** Use a block-based theme like `Didone`

- Open [the build from this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21544#issuecomment-1714309636)
- Open the editor
- Expect to see the default block appender with the right styling
- Tap on the default block appender
- **Expect** to see the new Paragraph block to match the font size of the default block appender
- Type some text
- **Expect** the value color to not have any opacity

## Screenshots or screencast 

**Note:** Screenshots done using the Didone theme.

Before Default Block Appender|After Default Block Appender
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/725ac952-28c5-43a3-b616-4324dff7b367" width=250 /></kbd>|<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/9616252d-2040-4bca-9871-af8ada71c4fc" width=250 /></kbd>

Before Placeholder color|After Placeholder color
-|-
<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/4685d76a-e799-4454-9f65-f5b7d38edba0" width=250 /></kbd>|<kbd><img src="https://github.com/WordPress/gutenberg/assets/4885740/34139cad-0705-4b39-be16-0daecbffcb87" width=250 /></kbd>
